src/assets/CollectiblesController.ts

```diff
@@ -798,56 +795,56 @@ export class CollectiblesController extends BaseController<
    */
   name = 'CollectiblesController';
 
-  private getAssetName: AssetsContractController['getAssetName'];
+  private getERC721AssetName: AssetsContractController['getERC721AssetName'];
```

Comment:
Just noticed this. Does CollectiblesController use AssetsContractController internally? If not, it seems odd that we would borrow another file's types like this. It feels like either we should have CollectiblesController inherit from AssetsContractController, or we should extract the functions that both CollectiblesController and AssetsContractController into an object so that that object can have a common type that is shared between the two controllers. As it stands, though, there is an implicit dependency on AssetsContractController that prevents AssetsContractController from changing without breaking this file.

Reply:
This is a common pattern throughout various controllers:
- https://github.com/MetaMask/controllers/blob/d7ec98b20cb52c62b776defe1fff4c1d632516b2/src/keyring/KeyringController.ts#L141
- https://github.com/MetaMask/controllers/blob/d7ec98b20cb52c62b776defe1fff4c1d632516b2/src/assets/TokenBalancesController.ts#L51
- https://github.com/MetaMask/controllers/blob/d7ec98b20cb52c62b776defe1fff4c1d632516b2/src/assets/AccountTrackerController.ts#L76

and it wasn't introduced in this PR in any case. Since this PR is already getting quite large and creeping scope can we address this concern in a followup? Or open a conversation about it in slack and go from there?

Reply:
cc @mcmire ☝️

Reply:
Okay, no problem, I didn't realize that. Sure, let's have a conversation in Slack about it. Other people may have some different ideas.